### PR TITLE
Native Animated - Add tests on iOS

### PR DESCRIPTION
**Pull request by janicduplessis**
Adds unit tests to the Native Animated implementation on iOS. This pretty much mirrors the tests we currently have on Android.

It also fixes 2 bugs I've found when adding the tests and pass the current time in `stepAnimation` instead of using `CACurrentMediaTime` to make testing easier.

- `stopListeningToAnimatedNodeValue` did not actually work at all, it should set the listener to nil.
- The finished value in the animation end callback was always true, this simplifies the `RCTAnimationDriver` interface to get rid of `removeAnimation` and fixes the end callback value.

**Test plan**
- Run the tests
- Make sure the UIExplorer example still works

### CODE REVIEW

File: Examples/UIExplorer/UIExplorer.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -727,11 +738,11 @@
 				14AADF041AC3DB95002390C9 /* libReact.a */,
 				2DD323D91DA2DD8B000FE1B8 /* libReact.a */,
 				3D3C08811DE3424E00C268FA /* libyoga.a */,
-				3D3C08831DE3424E00C268FA /* libyoga-tvOS.a */,
```
Comment (janicduplessis) [inline Examples/UIExplorer/UIExplorer.xcodeproj/project.pbxproj]: Not sure why these changes happen, didn't touch the tvOS target.

**Comment by janicduplessis** [comment]
cc @javache @ide

**Comment by janicduplessis** [comment]
@javache There was a conflict caused by my other pr landing, might have to re-import.

**Comment by javache** [comment]
It's landing now, I resolved the conflicts myself (you should verify when it lands)
